l1-contracts/test/foundry/l2/integration/L2GatewayTests.t.sol

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+// solhint-disable gas-custom-errors
+
+import {Test} from "forge-std/Test.sol";
+import "forge-std/console.sol";
+
+// import {BridgedStandardERC20} from "contracts/bridge/BridgedStandardERC20.sol";
+import {L2AssetRouter} from "contracts/bridge/asset-router/L2AssetRouter.sol";
+import {IL2NativeTokenVault} from "contracts/bridge/ntv/IL2NativeTokenVault.sol";
+
+import {UpgradeableBeacon} from "@openzeppelin/contracts-v4/proxy/beacon/UpgradeableBeacon.sol";
+// import {BeaconProxy} from "@openzeppelin/contracts-v4/proxy/beacon/BeaconProxy.sol";
+
+import {L2_ASSET_ROUTER_ADDR, L2_NATIVE_TOKEN_VAULT_ADDR, L2_BRIDGEHUB_ADDR} from "contracts/common/L2ContractAddresses.sol";
+import {ETH_TOKEN_ADDRESS, SETTLEMENT_LAYER_RELAY_SENDER} from "contracts/common/Config.sol";
+
+import {AddressAliasHelper} from "contracts/vendor/AddressAliasHelper.sol";
+import {BridgehubMintCTMAssetData} from "contracts/bridgehub/IBridgehub.sol";
+import {IAdmin} from "contracts/state-transition/chain-interfaces/IAdmin.sol";
+import {IL2AssetRouter} from "contracts/bridge/asset-router/IL2AssetRouter.sol";
+import {IL1Nullifier} from "contracts/bridge/interfaces/IL1Nullifier.sol";
+import {IL1AssetRouter} from "contracts/bridge/asset-router/IL1AssetRouter.sol";
+import {IBridgehub} from "contracts/bridgehub/IBridgehub.sol";
+
+import {L2Utils, SystemContractsArgs} from "./L2Utils.sol";
+
+import {SharedL2ContractL2DeployerUtils} from "./_SharedL2ContractL2DeployerUtils.sol";
+import {IChainTypeManager} from "contracts/state-transition/IChainTypeManager.sol";
+import {IZKChain} from "contracts/state-transition/chain-interfaces/IZKChain.sol";
+
+import {DeployUtils} from "deploy-scripts/DeployUtils.s.sol";
+import {SharedL2ContractL1DeployerUtils} from "../../l1/integration/l2-dummy-tests/_SharedL2ContractL1DeployerUtils.sol";
+import {L2GatewayTestAbstract} from "../../l1/integration/l2-dummy-tests/L2GatewayTestAbstract.t.sol";
+import {SharedL2ContractDeployer} from "../../l1/integration/l2-dummy-tests/_SharedL2ContractDeployer.sol";
+
+contract L2GatewayTests is Test, L2GatewayTestAbstract, SharedL2ContractL2DeployerUtils {
```

Comment:
it seems to be the same as `L2Erc20Test`

Reply:
the only difference is in what it imports